Automattic/jetpack
# VideoPress: Route video filters using query parameters

Fixes #26720.

## Proposed changes:


* Add support to three new query parameters: `rating`, `uploader` and `privacy`
* Set `rating`, `uploader` and `privacy` values on the query parameters when some filter value is checked
* React to changes on `rating`, `uploader` and `privacy` search params, so the filters on the client are set and the content gets filtered with the new filters
* Refactor the way we store the selected filters on the state, moving from objects to arrays, so handling the values become more simple
* PENDING: validate the values of the parameters

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Make sure you test on a site with multiple videos, so you can try multiple filters and can have filter results with multiple pages
* Go to `Jetpack > VideoPress`
* Apply some filter to the video results, for example Privacy = Private:

<img width="700" alt="Screen Shot 2023-01-12 at 17 41 34" src="https://user-images.githubusercontent.com/6760046/212176581-a8c8dd45-0dec-46ae-a5ea-022af2c49a12.png">

* Notice that the selected filter is now present on the query string portion of the URL, on the `privacy` parameter:

<img width="263" alt="Screen Shot 2023-01-12 at 17 43 02" src="https://user-images.githubusercontent.com/6760046/212176803-1c0a920b-bb8b-40e8-aa05-21f4649d2014.png">

* The same will happen with the Uploader and the Rating filters, that would produce the `uploader` and the `rating` parameters
* Notice that you can combine multiple filters and all of them will be present on the URL:

<img width="700" alt="Screen Shot 2023-01-12 at 17 44 50" src="https://user-images.githubusercontent.com/6760046/212177424-a5c0c8ea-0496-4349-b3cc-2758dba83d43.png">

* Notice that the pagination will be working as expected along with the filters, as well as the search field:

<img width="700" alt="Screen Shot 2023-01-12 at 17 48 52" src="https://user-images.githubusercontent.com/6760046/212178501-123cc383-c852-418b-baab-d51e3b2a4cd3.png">

* Apply some filters, copy the resulting URL and then open it using a new tab
   * Notice that the filter is applied to the videos
   * Nofice that the filter checkboxes corresponding to the filters from the URL are checked
* Try changing the filter values on URL
   * Notice that the videos get filtered by the new combination of filters
* Confirm that the application is working as expected, and you can delete, edit and upload videos as usual

**Known issues**

* the Go Back button from the Video Details page loses the filters, similiar to what happens for search; there is a followup issue related to it: #28260
